test_e2e.sh

```diff
@@ -194,6 +194,48 @@ fi
 RUNID="${RANDOM}${RANDOM}"
 DIR="/tmp/git-sync-e2e.$RUNID"
 mkdir "$DIR"
+function final_cleanup() {
+  if [[ "${CLEANUP:-}" == 0 ]]; then
+      echo "leaving logs in $DIR"
+  else
+      rm -rf "$DIR"
+  fi
+}
+# Set the trap to call the final_cleanup function on exit.
+trap final_cleanup EXIT
+
+skip_github_app_test="${SKIP_GITHUB_APP_TEST:-false}"
+required_env_vars=()
+LOCAL_GITHUB_APP_PRIVATE_KEY_FILE="github_app_private_key.pem"
+GITHUB_APP_PRIVATE_KEY_MOUNT=""
```

Comment:
This should be declared as an empty array so that it is properly expanded below when the volume mount is not added.

e.g.
```bash
GITHUB_APP_PRIVATE_KEY_MOUNT=()
...
  # Mount the GitHub App private key file to the git-sync container
  GITHUB_APP_PRIVATE_KEY_MOUNT+=(-v "${TEST_GITHUB_APP_PRIVATE_KEY_FILE}:/${LOCAL_GITHUB_APP_PRIVATE_KEY_FILE}:ro")
...
        "${GITHUB_APP_PRIVATE_KEY_MOUNT[@]}" \
        "${GIT_SYNC_E2E_IMAGE}" \
```

Reply:
Changing it to an empty array looks good to me. Thanks for pointing it out.